File: wiki/versions/1.19/item/simple_item/java/src_mojmaps/main/java/org/quiltmc/wiki/simple_item/SimpleItemExample.java
package org.quiltmc.wiki.simple_item;

import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.Item;
import org.quiltmc.loader.api.ModContainer;
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer;
import org.quiltmc.qsl.item.setting.api.QuiltItemSettings;

public class SimpleItemExample implements ModInitializer {

	// @start Declaration
	public static final Item EXAMPLE_ITEM = new Item(new QuiltItemSettings().tab(CreativeModeTab.TAB_MISC));
	// @end Declaration

	@Override
	public void onInitialize(ModContainer mod) {
		// @start Registration
		Registry.register(Registry.ITEM, new ResourceLocation(mod.metadata().id(), "example_item"), EXAMPLE_ITEM);
		// @end Registration
	}
}

File: wiki/versions/1.19/item/simple_item/kotlin/src/main/kotlin/org/quiltmc/wiki/simple_item/SimpleItemExample.kt
package org.quiltmc.wiki.simple_item

import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraft.util.registry.Registry
import org.quiltmc.loader.api.ModContainer
import org.quiltmc.qkl.wrapper.minecraft.registry.registryScope
import org.quiltmc.qkl.wrapper.qsl.items.itemSettingsOf
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer

// @start Declaration
val EXAMPLE_ITEM: Item = Item(itemSettingsOf(group = ItemGroup.MISC))
// @end Declaration

object SimpleItemExample: ModInitializer {

    override fun onInitialize(mod: ModContainer) {
        // @start Registration
        registryScope(mod.metadata().id()) {
            EXAMPLE_ITEM withPath "example_item" toRegistry Registry.ITEM
        }
        // @end Registration
    }
}
